rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,110 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie as incidents are intended to be used for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured in Teleport's config yaml file. The required fields will be added to a 'plugins' section containing the required information to interact with both Teleport access and the Opsgenie API.
+
+```
+plugins:
+    opsgenie:
+        api_key: "path/to/key" # File containing Opsgenie API Key
+        opsgenie_addr: "example.app.opsgenie.com" # Address of Opsgenie
+        priority: "2" # Priority to create Opsgenie alerts with
+        alert_tags: ["example-tag"] # List of tags to be added to alerts created in Opsgenie
+        auto_approval: true # Whether or not to enable auto approval
```

Comment:
For PagerDuty plugin auto-approval is done by having an annotation on the user's role that specifies services the user is on-call in. Why do we need this flag for OpsGenie?

Reply:
Sorry, this was probably added because I commented that auto-approval should be configurable.
PagerDuty AutoApproval is not configurable, so feel free to ignore my previous comment.

Reply:
I'll remove this flag then and handle this the same way pagerduty plugin does